### ISSUE TITLE
Dependency problem in CMake when compiling Fortran examples

### ISSUE THREAD

**Issue by havogt** [eth-cscs/serialbox2#47]
Sometimes the compilation fails with the following error

```
Fatal Error: Can't delete temporary module file './m_ser.mod0': No such file or directory
Fatal Error: Can't delete temporary module file './m_ser.mod0': No such file or directory
make[3]: *** [CMakeFiles/main_consumer.dir/pp/m_ser.f90.o] Error 1
make[3]: *** [CMakeFiles/main_consumer_perturb.dir/pp/m_ser.f90.o] Error 1
make[2]: *** [CMakeFiles/main_consumer.dir/pp/m_ser.f90.o.provides] Error 2
make[2]: *** [CMakeFiles/main_consumer_perturb.dir/pp/m_ser.f90.o.provides] Error 2
make[1]: *** [CMakeFiles/main_consumer.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/main_consumer_perturb.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by havogt** [comment]
m_serialize.f90 should go into its one (object) library to properly setup the dependency.

**Comment by havogt** [comment]
Fixed in #179
